### DISCUSSION TITLE
Is there a shorter method in SeleniumBase to handle waiting for elements

### DISCUSSION THREAD

**Discussion by joako233** [seleniumbase/SeleniumBase#2739] [Q&A] [answered]
Hello SeleniumBase community,

I'm currently using SeleniumBase for my testing automation tasks, and I frequently find myself using the EC.presence_of_all_elements_located((By.XPATH,"//h2[@class='h4']/..")) method to wait for the presence of multiple elements on a page.

However, I was wondering if there's a shorter or more concise method within SeleniumBase that accomplishes the same task? While the current method works well, I'm always interested in streamlining my code and making it more efficient.

Any suggestions or insights would be greatly appreciated!

Thanks!!
![image](https://github.com/seleniumbase/SeleniumBase/assets/141453948/7f61c2c8-1997-4901-a94e-d57cc2109642)

**Comment by mdmintz** [comment, marked as answer]
For returning a list of elements:

```python
elements = driver.find_elements("h2")
```

(It autodetects between CSS and XPath)

For waiting on the first occurrence: `driver.wait_for_element`. Eg:

```python
driver.wait_for_element("h2")
```

**Reply by joako233** [reply]
But when I try to iterate between the elements, this message appears:
![image](https://github.com/seleniumbase/SeleniumBase/assets/141453948/17488970-bee0-4462-a9ca-62e9f462b30f)

**Reply by mdmintz** [reply]
For the list:

```python
elements = driver.find_elements("h2")

for element in elements:
    # ...
```

**Comment by joako233** [comment]
Thanks Michael!
